### PR TITLE
ASCII Schedule Formatter

### PR DESCRIPTION
**Pull request by EclipseIsDead**
# Description
ASCII table around each session schedule, using early and late to bound the schedule height.

Ideas:
- Colour?
- More descriptive entries?

## Type of change
- [x] New feature 
- [x] Breaking change 
- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] ASCIIFormatterTest (high method coverage)

# Does this PR follow the rubric?

Yes.

# Checklist:

- [x] I ran `./gradlew spotlessApply`
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I ran `./gradlew build` and the gradle build succeeded
- [x] I have not violated any design principles

### CODE REVIEW

File: src/main/java/entities/ASCIIFormatter.java [hunk 1]
```diff
@@ -11,5 +16,55 @@
...
+    private String genBorder() {
+        String leftAlignFormat = "| %-15s | %-4d |%n";
+        StringBuilder output = new StringBuilder("Schedule: \n\n");
+
+        // The header for our schedule
+        output.append("+----------------+----------------+----------------+----------------+----------------+%n" +
+                "|     Monday     |     Tuesday    |    Wednesday   |    Thursday    |     Friday     |%n" +
+                "+----------------+----------------+----------------+----------------+----------------+%n");
+        for (int i = start.getHour(); i < end.getHour(); i++) {
+            output.append(String.format(leftAlignFormat, "PogU" + i, i * i));
```
Comment (sugolov) [inline src/main/java/entities/ASCIIFormatter.java:64]: what does "PogU" do lol

Comment (EclipseIsDead) [reply src/main/java/entities/ASCIIFormatter.java]: temp holder for da string

**Comment by EclipseIsDead** [comment]
Yo what's up with `SchedulerTest`?

**Review by sugolov — CHANGES_REQUESTED** [review]
overall looks good. please add course names to the schedule slots though. also, how does the ascii handle schedules with conflicts?

File: src/main/java/entities/ASCIIFormatter.java [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+     * @returnlist of timeslots
+     */
+    public Set<Timeslot> getTimeslots() {
+        return timeslots;
+    }
+
+    /**
+     * Check which sessions we need for however many schedules
+     *
+     * @return list of Booleans (0 is for F, 1 is for F)
```
Comment (kenneth-miura) [inline src/main/java/entities/ASCIIFormatter.java:70]: I'm assuming 1 is for S lol

**Review by kenneth-miura — CHANGES_REQUESTED** [review]
1 typo in docs, but besides that looks great

File: src/main/java/entities/ASCIIFormatter.java (hunk 2 above)
Comment (EclipseIsDead) [reply src/main/java/entities/ASCIIFormatter.java:70]: lol

**Pull request merged**